File: email-archive/email_archive/markdown_formatter.py
"""Markdown formatting utilities for export commands."""

from __future__ import annotations

from datetime import datetime

from .store import ContactExport, ReviewPeriodExport


def format_contact_section(contact: ContactExport) -> str:
    """Format a contact export as a markdown section."""
    header = contact.name if contact.name else contact.email
    lines = [f"### {header}", ""]

    if contact.name:
        lines.append(f"**Email:** {contact.email}")

    lines.append(
        f"**Communication:** {contact.total_emails} total emails "
        f"({contact.sent_to} sent, {contact.received_from} received)"
    )
    lines.append(f"**Direction:** {contact.communication_direction}")
    lines.append(f"**First Contact:** {contact.first_contact:%Y-%m-%d}")
    lines.append(f"**Last Contact:** {contact.last_contact:%Y-%m-%d}")
    lines.append("")

    return "\n".join(lines) + "\n"


def format_review_email_section(review: ReviewPeriodExport) -> str:
    """Format a review period export as a markdown email activity section."""
    lines = [
        "## Email Activity",
        "",
        f"**Period:** {review.period_start:%Y-%m-%d} to {review.period_end:%Y-%m-%d}",
        "",
        "### Summary",
        f"- **Total Emails:** {review.email_count}",
        f"- **Sent:** {review.sent_count}",
        f"- **Received:** {review.received_count}",
        f"- **Peak Activity:** {review.peak_activity_day} at {review.peak_activity_hour}:00",
        "",
    ]

    if review.top_contacts:
        lines.append("### Top Contacts")
        for contact in review.top_contacts:
            name = contact.name if contact.name else contact.email
            lines.append(f"- **{name}** ({contact.email}): {contact.total_emails} emails")
        lines.append("")

    return "\n".join(lines) + "\n"


def format_ideas_header(title: str, date: datetime, status: str) -> str:
    """Format a header for ideas/thoughts files."""
    lines = [
        f"# {title}",
        "",
        f"**Added:** {date:%Y-%m-%d}",
        f"**Status:** {status}",
        "**Related:** ",
        "",
    ]
    return "\n".join(lines) + "\n"

File: email-archive/src/EmailArchive/Export/MarkdownFormatter.cs
// src/EmailArchive/Export/MarkdownFormatter.cs
using System.Text;

namespace EmailArchive.Export;

/// <summary>
/// Static utility class for formatting export data as markdown.
/// </summary>
public static class MarkdownFormatter
{
    /// <summary>
    /// Formats a contact export as a markdown section.
    /// </summary>
    /// <param name="contact">The contact to format.</param>
    /// <returns>Markdown formatted contact section.</returns>
    public static string FormatContactSection(ContactExport contact)
    {
        var sb = new StringBuilder();

        // Use name as header, or email if name is empty
        var header = string.IsNullOrWhiteSpace(contact.Name) ? contact.Email : contact.Name;
        sb.AppendLine($"### {header}");
        sb.AppendLine();

        // Email (only if we used name as header)
        if (!string.IsNullOrWhiteSpace(contact.Name))
        {
            sb.AppendLine($"**Email:** {contact.Email}");
        }

        // Communication stats
        sb.AppendLine($"**Communication:** {contact.TotalEmails} total emails ({contact.SentTo} sent, {contact.ReceivedFrom} received)");
        sb.AppendLine($"**Direction:** {contact.CommunicationDirection}");
        sb.AppendLine($"**First Contact:** {contact.FirstContact:yyyy-MM-dd}");
        sb.AppendLine($"**Last Contact:** {contact.LastContact:yyyy-MM-dd}");
        sb.AppendLine();

        return sb.ToString();
    }

    /// <summary>
    /// Formats a review period export as a markdown email activity section.
    /// </summary>
    /// <param name="review">The review period data to format.</param>
    /// <returns>Markdown formatted email activity section.</returns>
    public static string FormatReviewEmailSection(ReviewPeriodExport review)
    {
        var sb = new StringBuilder();

        sb.AppendLine("## Email Activity");
        sb.AppendLine();
        sb.AppendLine($"**Period:** {review.PeriodStart:yyyy-MM-dd} to {review.PeriodEnd:yyyy-MM-dd}");
        sb.AppendLine();

        // Summary stats
        sb.AppendLine("### Summary");
        sb.AppendLine($"- **Total Emails:** {review.EmailCount}");
        sb.AppendLine($"- **Sent:** {review.SentCount}");
        sb.AppendLine($"- **Received:** {review.ReceivedCount}");
        sb.AppendLine($"- **Peak Activity:** {review.PeakActivityDay} at {review.PeakActivityHour}:00");
        sb.AppendLine();

        // Top contacts
        if (review.TopContacts.Count > 0)
        {
            sb.AppendLine("### Top Contacts");
            foreach (var contact in review.TopContacts)
            {
                var name = string.IsNullOrWhiteSpace(contact.Name) ? contact.Email : contact.Name;
                sb.AppendLine($"- **{name}** ({contact.Email}): {contact.TotalEmails} emails");
            }
            sb.AppendLine();
        }

        return sb.ToString();
    }

    /// <summary>
    /// Formats a header for ideas/thoughts files in the areas/ directory.
    /// </summary>
    /// <param name="title">The title of the idea.</param>
    /// <param name="date">The date the idea was added.</param>
    /// <param name="status">The status (seed, developing, mature, archived).</param>
    /// <returns>Markdown formatted header section.</returns>
    public static string FormatIdeasHeader(string title, DateTime date, string status)
    {
        var sb = new StringBuilder();

        sb.AppendLine($"# {title}");
        sb.AppendLine();
        sb.AppendLine($"**Added:** {date:yyyy-MM-dd}");
        sb.AppendLine($"**Status:** {status}");
        sb.AppendLine("**Related:** ");
        sb.AppendLine();

        return sb.ToString();
    }
}
